packages/adapters/src/kubernetes-client-adapter.ts

```diff
@@ -0,0 +1,140 @@
+        this.logger = new ObjLogger(this.name);
+        this._configPath = configPath;
+        this._namespace = namespace;
+    }
+
+    private get config(): k8s.KubeConfig {
+        if (!this._config) {
+            const kc = new k8s.KubeConfig();
+
+            try {
+                if (this._configPath && this._configPath.length) {
+                    kc.loadFromFile(this._configPath);
+                } else {
+                    kc.loadFromCluster();
+                }
+
+                this._config = kc;
+            } catch (err: any) {
+                this.logger.error("Unable to load kubeconfig", err);
+            }
+        }
+
+        if (!this._config) {
+            throw new Error("Kubernetes API client not initialized");
+        }
+
+        return this._config;
+    }
+
+    async createPod(metadata: k8s.V1ObjectMeta, spec: k8s.V1PodSpec, retries: number = 0) {
+        const kubeApi = this.config.makeApiClient(k8s.CoreV1Api);
+
+        const result = await this.runWithRetries(retries, "Create Pod", async () => {
+            return await kubeApi.createNamespacedPod(this._namespace, {
+                apiVersion: "v1",
+                kind: "Pod",
+                metadata,
+                spec
+            });
+        });
+
+        return result as {
+            response: http.IncomingMessage;
+            body: k8s.V1Pod;
+        };
+    }
+
+    async deletePod(podName: string, retries: number = 0) {
+        const kubeApi = this.config.makeApiClient(k8s.CoreV1Api);
+
+        const result = await this.runWithRetries(retries, "Delete Pod", () =>
+            kubeApi.deleteNamespacedPod(podName, this._namespace, undefined, undefined, 0)
+        );
+
+        return result as {
+            response: http.IncomingMessage;
+            body: k8s.V1Pod;
+        };
+    }
+
+    async exec(podName: string, containerName: string, command: string | string[],
+        stdout: Writable | null, stderr: Writable | null, stdin: Readable | null, retries: number = 0
+    ) {
+        const exec = new k8s.Exec(this.config);
+
+        await this.runWithRetries(retries, "Exec", () =>
+            exec.exec(this._namespace, podName, containerName, command, stdout, stderr, stdin, false,
+                (...args) => this.logger.debug("exec status", ...args))
+        );
+    }
+
+    async waitForPodStatus(podName: string, expectedStatuses: string[]): Promise<string> {
+        const kubeApi = this.config.makeApiClient(k8s.CoreV1Api);
+
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+            try {
+                const response = await kubeApi.readNamespacedPodStatus(podName, this._namespace);
+                const status = response.body.status?.phase || "";
+
+                if (expectedStatuses.includes(status)) {
+                    return status;
+                }
+            } catch (err: any) {
+                this.logger.error("Failed to get pod status", err);
```

Comment:
Won't this fall into an infinite loop and just spam the system with log message twice every second?

Reply:
fixed